packages/components/src/query-controls/index.js

```diff
@@ -66,7 +69,16 @@ export default function QueryControls( {
 				} }
 			/>
 		),
-		onCategoryChange && (
+		categoriesList && onCategoryChange && (
+			<CategorySelect
+				key="query-controls-category-select"
+				categoriesList={ categoriesList }
+				label={ __( 'Category' ) }
+				noOptionLabel={ __( 'All' ) }
+				selectedCategoryId={ selectedCategoryId }
```

Comment:
@hypest and @pinarol, there were some mobile-related props in the original commit:
https://github.com/WordPress/gutenberg/commit/e17d104ee3fe6eb8d81373226ab73096f59f9ba3

Do you think `MOBILE_CONTROL_PROPS` is still needed? Can someone double-check that?

Reply:
👋 @chipsnyder , can you have a look here since you worked on the Latest Posts block in the past? Thanks!

Reply:
In the current state, we don't really need `MOBILE_CONTROL_PROPS` added back. 

Because of the original change we did end up removing categories and hiding them behind a `__DEV__` flag. https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/latest-posts/edit.native.js#L181

<details>
<summary>With this change, we could reenable categories on that block but there would be two issues we might want to consider.</summary>

The current Native implementation doesn't handle:
-  Multiselect
- For a long list of categories, the control that was used by default didn't handle scroll

We have https://github.com/wordpress-mobile/gutenberg-mobile/issues/2069 to track that work.
</details>

So I'd say we're good here but does raise the question again of if we want to keep categories disabled in mobile or not. 

ps thanks for checking with use @gziolo!